addonscript/utils.py

```diff
@@ -0,0 +1,93 @@
+"""Common utilities for addonscript."""
+
+import time
+from uuid import uuid4
+
+from jose import jws
+from async_timeout import timeout
+
+
+def generate_JWT(context, _iat=None):
+    """Create and Sign a JWT valid for AMO HTTP requests.
+
+    Uses context.config 'jwt_user' and 'jwt_secret' and sets an expiration of
+    4 minutes (AMO supports a max of 5)
+    """
+    user = context.config['jwt_user']
+    secret = context.config['jwt_secret']
+    jti = str(uuid4())
+    iat = _iat or int(time.time())  # _iat for testing sanity
+    exp = iat + 60*4  # AMO has a 5 minute max, so set this to 4 minutes after issued
+    payload = {
+        'iss': user,
+        'jti': jti,
+        'iat': iat,
+        'exp': exp,
+    }
+    token_str = jws.sign(payload, secret, algorithm='HS256')
+    return token_str
+
+
+async def amo_get(context, url):
+    """Perform a GET request against AMO's API.
+
+    Automatically fills in the HTTP header with the Authorization token.
+    Assumes request will return a valid json object, on success.
+    """
+    async with timeout(30):
+        resp = context.session.get(
+            url, headers={
+                'Authorization': 'JWT {}'.format(generate_JWT(context))
+                },
+            )
+        async with resp as r:
+            r.raise_for_status()
```

Comment:
We're using `raise_for_status` to make sure we have a good response, which is good. However, scriptworker gets the failure/exception info by the exit status of the *script... the pattern we've been using is to catch errors and raise the appropriate `ScriptWorkerTaskException`, so we can more easily differentiate between network errors, server errors, script failures, and bad task definitions.